packages/@lwc/engine-dom/src/apis/hydrate-component.ts

```diff
@@ -38,14 +33,12 @@ export function hydrateComponent(
         );
     }
 
-    const def = getComponentInternalDef(Ctor);
-
     try {
         // Let the renderer know we are hydrating, so it does not replace the existing shadowRoot
         // and uses the same algo to create the stylesheets as in SSR.
         setIsHydrating(true);
 
-        createVM(element, def, {
+        createVM(element, Ctor, {
```

Comment:
Moving the `getComponentInternalDef` inside the try-catch block makes the [this test](https://github.com/salesforce/lwc/blob/438323a485f3e18245f1c934e8cf963c0fde795a/packages/integration-karma/test/hydration/index.spec.js#L12-L20) to fail:
- Before this change, the `hydrateComponent` method throws an error outside the try-catch block and stop its execution
- With this change, the error is thrown inside the try-catch block, and hydration attempts to recover from it without success. Invoking `createElement` will run into the same issue.

@jodarove What do you think about moving the `createVM` outside the try-catch block?

Reply:
I'm fine doing that. what I don't like is that does not differentiate from our own exceptions and errors thrown in the component (ex: the constructor), but I guess the same error may be thrown again when invoking the `createElement`.

Another alternative is to verify that the `Ctor` is a valid component constructor.

Reply:
> I don't like is that does not differentiate from our own exceptions and errors thrown in the component (ex: the constructor), but I guess the same error may be thrown again when invoking the `createElement`.

Overall I think, we are not there yet in terms of error handling in rehydration. This is something we can improve in the future based on consumers' feedback. In the meantime, I updated the test to check the error logs instead of looking at the exception. Let me know what you think.